server/src/main/java/org/elasticsearch/cluster/coordination/UnsafeBootstrapMasterCommand.java

```diff
@@ -0,0 +1,187 @@
+        if (nodeMetaData == null) {
+            throw new ElasticsearchException(NO_NODE_METADATA_FOUND_MSG);
+        }
+
+        String nodeId = nodeMetaData.nodeId();
+        terminal.println(Terminal.Verbosity.VERBOSE, "Current nodeId is " + nodeId);
+        terminal.println(Terminal.Verbosity.VERBOSE, "Loading manifest file");
+        final Manifest manifest = Manifest.FORMAT.loadLatestState(logger, namedXContentRegistry, dataPaths);
+
+        if (manifest == null) {
+            throw new ElasticsearchException(NO_MANIFEST_FILE_FOUND_MSG);
+        }
+        if (manifest.isGlobalGenerationMissing()) {
+            throw new ElasticsearchException(GLOBAL_GENERATION_MISSING_MSG);
+        }
+        terminal.println(Terminal.Verbosity.VERBOSE, "Loading global metadata file");
+        final MetaData metaData = MetaData.FORMAT.loadGeneration(logger, namedXContentRegistry, manifest.getGlobalGeneration(),
+                dataPaths);
+        if (metaData == null) {
+            throw new ElasticsearchException(NO_GLOBAL_METADATA_MSG + " [generation = " + manifest.getGlobalGeneration() + "]");
+        }
+        final CoordinationMetaData coordinationMetaData = metaData.coordinationMetaData();
+        if (coordinationMetaData == null ||
+                coordinationMetaData.getLastCommittedConfiguration() == null ||
+                coordinationMetaData.getLastCommittedConfiguration().isEmpty()) {
+            throw new ElasticsearchException(EMPTY_LAST_COMMITTED_VOTING_CONFIG_MSG);
+        }
+
+        CoordinationMetaData newCoordinationMetaData = CoordinationMetaData.builder(coordinationMetaData)
+                .clearVotingConfigExclusions()
+                .lastAcceptedConfiguration(new CoordinationMetaData.VotingConfiguration(Collections.singleton(nodeId)))
+                .lastCommittedConfiguration(new CoordinationMetaData.VotingConfiguration(Collections.singleton(nodeId)))
+                .build();
+        terminal.println(Terminal.Verbosity.VERBOSE, "New coordination metadata is constructed " + newCoordinationMetaData);
+        Settings persistentSettings = Settings.builder()
+                .put(metaData.persistentSettings())
+                .put(UNSAFE_BOOTSTRAP.getKey(), true)
+                .build();
+        MetaData newMetaData = MetaData.builder(metaData)
+                .persistentSettings(persistentSettings)
+                .coordinationMetaData(newCoordinationMetaData)
+                .build();
+        writeNewMetaData(terminal, manifest, newMetaData, dataPaths);
+    }
+
+    private void writeNewMetaData(Terminal terminal, Manifest manifest, MetaData newMetaData, Path[] dataPaths) {
+        try {
+            terminal.println(Terminal.Verbosity.VERBOSE, "Writing new global metadata to disk");
+            long newGeneration = MetaData.FORMAT.write(newMetaData, dataPaths);
+            Manifest newManifest = new Manifest(manifest.getCurrentTerm(), manifest.getClusterStateVersion(), newGeneration,
```

Comment:
I think we should increment the term, and log that we've done so.

Reply:
I've implemented increment in this commit https://github.com/elastic/elasticsearch/pull/37696/commits/0e5091330717968d54887d3f6db7c37b1c3236cc. For now, I'm logging new `currentTerm` to the terminal, will get back to it - once I figure out how our command line tools work with elasticsearch logs.